openmeter/billing/service/invoice.go

```diff
@@ -828,6 +834,27 @@ func (s *Service) UpdateInvoice(ctx context.Context, input billing.UpdateInvoice
 	)
 }
 
+// updateInvoice calls the adapter to update the invoice and returns the updated invoice including any expands that are
+// the responsibility of the service
+func (s Service) updateInvoice(ctx context.Context, in billing.UpdateInvoiceAdapterInput) (billing.Invoice, error) {
+	invoice, err := s.adapter.UpdateInvoice(ctx, in)
+	if err != nil {
+		return billing.Invoice{}, err
+	}
+
+	invoice, err = s.resolveWorkflowApps(ctx, invoice)
+	if err != nil {
+		return billing.Invoice{}, fmt.Errorf("error adding fields to invoice [%s]: %w", invoice.ID, err)
+	}
+
+	invoice, err = s.resolveStatusDetails(ctx, invoice)
```

Comment:
TODO: in most cases this will be resolved once more, so let's optimize it if possible